### DISCUSSION TITLE
Homing Issues

### DISCUSSION THREAD

**Discussion by JC286006** [grblHAL/iMXRT1062#64] [General]
Hello Everyone 
 I have built my cnc router and used one of Phil Barrett`s BOB. It would home at first now every time I try to do a homing it moves to the limit switches and then sends an alarm and stops all movement until I manually move the axis off of the limit switch reset and clear the alarm. Anyone have an idea as to what could be causing this. I am attaching a pic of the settings in IoSender 
![IMG_20230807_101755322_HDR](https://github.com/grblHAL/iMXRT1062/assets/141972434/36c1d2a2-1f36-4f42-b963-244e842e25f2)
![IMG_20230807_102218957_HDR](https://github.com/grblHAL/iMXRT1062/assets/141972434/cbec84b1-2926-4b4f-a0c2-c08085921660)

**Comment by terjeio** [comment]
Your settings for dual axis homing ($347-349) are too small or the second limit switch is not working?
The _Signals_ LED for the auto squared axis should lit up when either of its limit switches are triggered. Does it?

**Comment by JC286006** [comment]
Hello Terjeio
 I have checked all limit switches when pressed they all send an alarm to the correct axis alarm icon so they are working correctly.
 I have six limit switches installed to prevent the machine from crashing at both ends of movement on all three axis to prevent over travel. 
 On the Y axis which is the auto squared axis if either limit is touched at the homing end of travel it sends an alarm.
 I have changed the dual axis homing to 5mm from 0mm and back to 0mm still no change all axis go to the limit then signal an alarm.
 I just don`t understand this the homing was working the only changes that were made were to fine tune steps per rev, max feed rate, and the work area size  I didn`t change anything else,

**Reply by terjeio** [reply]
> ... used one of Phil Barrett`s BOB

Which?

Y limit switches are swapped (not associated with the correct motor)?
When the alarm happens are both switches triggered? If you send the `$LIM` command you can see the status of each switch separately.
To track what happens I suggest you home Y separately in the first pass, then X and Z.

**Comment by JC286006** [comment]
I bought this board from Phil the T41U5XBB with ethernet +USB assembly kit . 
I get alarms on all axis when homing and all switches look to be triggered when the alarms are thrown.
The  Y switches look to both be triggered when there is an alarm I will check them with my meter.
The Y limits could be on the wrong axis limit connection,  if they are I will correct then retry. 
Why would it home then stop homing when no changes were made to wiring etc. ?
I will change the homing and see what happens and send the $LIM command.
Thank you for your help sorting this out James
    On Friday, August 11, 2023 at 12:58:13 PM CDT, Terje Io ***@***.***> wrote:  
 
 



... used one of Phil Barrett`s BOB


Which?

Y limit switches are swapped (not associated with the correct motor)?
When the alarm happens are both switches triggered? If you send the $LIM command you can see the status of each switch separately.
To track what happens I suggest you home Y separately in the first pass, then X and Z.

**Comment by JC286006** [comment]
I changed the homing to y first pass x second pass z third pass now homing is working!!!!
I do have an issue I take the z axis to its lowest position and zero it.
 Then I hit the home button when homing completes it moves down the 5mm but the DRO says 76.001 it should be saying 92.000. The  Z has.97mm of movement from bottom point to the top point.
When I do an MDI command of G0 Z0 it goes down to 0.000 on the DRO but Z still has 20mm of downward movement to go before it actually is at the 0 point of downward movement.
I have reset 0 three times on the z axis and rehomed each time it has went to the limit done the homing moves correctly then when complete the DRO is showing 76.001 after the 5mm back off it should be showing 92.000.
Do I have a setting not correct to cause this DRO reading after homing?   
    On Friday, August 11, 2023 at 01:33:17 PM CDT, James Coleman ***@***.***> wrote:  
 
  I bought this board from Phil the T41U5XBB with ethernet +USB assembly kit . 
I get alarms on all axis when homing and all switches look to be triggered when the alarms are thrown.
The  Y switches look to both be triggered when there is an alarm I will check them with my meter.
The Y limits could be on the wrong axis limit connection,  if they are I will correct then retry. 
Why would it home then stop homing when no changes were made to wiring etc. ?
I will change the homing and see what happens and send the $LIM command.
Thank you for your help sorting this out James
    On Friday, August 11, 2023 at 12:58:13 PM CDT, Terje Io ***@***.***> wrote:  
 
 



... used one of Phil Barrett`s BOB


Which?

Y limit switches are swapped (not associated with the correct motor)?
When the alarm happens are both switches triggered? If you send the $LIM command you can see the status of each switch separately.
To track what happens I suggest you home Y separately in the first pass, then X and Z.

**Reply by terjeio** [reply]
I have retested the driver with my simulator and homing works as it should even with all three axes homed at the same time. It is a bit unusual to do that, normally one wants the Z to home first (to the upper position) to get it out of the way of any obstacles.

> I just dont understand this the homing was working the only changes that were made were to fine tune steps per rev, max feed rate, and the work area size

The work area size is used to calculate the dual axis fail distance (from the $347 setting) so this may have an effect on when Alarm 15 is raised.

> I have reset 0 three times on the z axis and rehomed each time it has went to the limit done the homing moves correctly then when complete the DRO is showing 76.001 after the 5mm back off it should be showing 92.000.

The DRO shows the work position including any offsets. What is shown if you clear any offsets in the _Offsets_ tab?

You may also enable _Show text overlay_ in the 3D viewer config in the _Settings: App_ tab to get both the machine and work positions overlaid the 3D view. Or use the _MPos_ flyout if you just want to briefly check the machine position.

Finally you can force the home position to 0,0,0 by checking _Set machine origin to 0_ in the $22 _Homing cycle_ setting. The default is to set the home position to negative max travel configuration.

**Comment by JC286006** [comment]
Hello Terje Lo I finally have everything working Thank You for your time and help James

    On Friday, August 11, 2023 at 01:33:17 PM CDT, James Coleman ***@***.***> wrote:  
 
  I bought this board from Phil the T41U5XBB with ethernet +USB assembly kit . 
I get alarms on all axis when homing and all switches look to be triggered when the alarms are thrown.
The  Y switches look to both be triggered when there is an alarm I will check them with my meter.
The Y limits could be on the wrong axis limit connection,  if they are I will correct then retry. 
Why would it home then stop homing when no changes were made to wiring etc. ?
I will change the homing and see what happens and send the $LIM command.
Thank you for your help sorting this out James
    On Friday, August 11, 2023 at 12:58:13 PM CDT, Terje Io ***@***.***> wrote:  
 
 



... used one of Phil Barrett`s BOB


Which?

Y limit switches are swapped (not associated with the correct motor)?
When the alarm happens are both switches triggered? If you send the $LIM command you can see the status of each switch separately.
To track what happens I suggest you home Y separately in the first pass, then X and Z.
